internal/etl/registry/pipe/event_log.go

```diff
@@ -0,0 +1,157 @@
+
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// NewEventParserPipe ... Initializer
+func NewEventParserPipe(ctx context.Context, cfg *core.ClientConfig,
+	opts ...component.Option) (component.Component, error) {
+	client := client.NewEthClient()
+
+	ed := &EventDefinition{
+		cfg:    cfg,
+		client: client,
+	}
+
+	return component.NewPipe(ctx, ed, core.GethBlock, core.EventLog, opts...)
+}
+
+// contractEvents ... Struct to hold the contract address and the event signatures
+type contractEvents struct {
+	address common.Address
+	sigs    []common.Hash
+}
+
+// HasSignature ... Checks if the event has the signature
+func (ce *contractEvents) HasSignature(sig common.Hash) bool {
+	for _, s := range ce.sigs {
+		if s == sig {
+			return true
+		}
+	}
+
+	return false
+}
+
+// getEventsToMonitor ... Gets the smart contract events to monitor from the state store
+func (ed *EventDefinition) getEventsToMonitor(ctx context.Context, rt core.RegisterType,
+	addresses []string, ss state.Store) ([]contractEvents, error) {
+	var events []contractEvents
+	for _, address := range addresses {
+		addrKey := state.MakeKey(rt, address, false).WithPUUID(ed.pUUID)
+		sigs, err := ss.GetSlice(ctx, addrKey)
+		if err != nil {
+			logging.WithContext(ctx).Error(err.Error())
+			return []contractEvents{}, err
+		}
+
+		var parsedSigs []common.Hash
+		for _, sig := range sigs {
+			parsedSigs = append(parsedSigs, crypto.Keccak256Hash([]byte(sig)))
+		}
+
+		logging.WithContext(ctx).Debug(fmt.Sprintf("Address: %s, Sigs: %v", address, parsedSigs))
+		events = append(events, contractEvents{
+			address: common.HexToAddress(address),
+			sigs:    parsedSigs,
+		})
+	}
+
+	return events, nil
+}
+
+// Transform ... Gets the events from the block, filters them and
+// returns them if they are in the list of events to monitor
+func (ed *EventDefinition) Transform(ctx context.Context, td core.TransitData) ([]core.TransitData, error) {
+	asBlock, success := td.Value.(types.Block)
+	if !success {
+		return []core.TransitData{}, fmt.Errorf("could not convert to block")
+	}
+
+	stateStore, err := state.FromContext(ctx)
+	if err != nil {
+		return []core.TransitData{}, err
+	}
+
+	logging.NoContext().Debug("Getting addresess",
+		zap.String(core.PUUIDKey, ed.pUUID.String()))
+
+	addresses, err := stateStore.GetSlice(ctx, ed.sk)
+	if err != nil {
+		return []core.TransitData{}, err
+	}
+
+	eventsToMonitor, err := ed.getEventsToMonitor(ctx, core.EventLog, addresses, stateStore)
+	if err != nil {
+		return []core.TransitData{}, err
+	}
+
+	hash := asBlock.Header().Hash()
+
+	query := ethereum.FilterQuery{
+		BlockHash: &hash,
+		Addresses: pess_common.SliceToAddresses(addresses),
+	}
+
+	logs, err := ed.client.FilterLogs(context.Background(), query)
+	if err != nil {
+		return []core.TransitData{}, err
+	}
+
+	result := make([]core.TransitData, 0)
+	for _, log := range logs {
+		for _, event := range eventsToMonitor {
```

Comment:
Considering the logs filtered aren't that many, that this double loop isn't that much of a problem right ?

Reply:
Ahhh great catch, yes and no. There shouldn't be too many operations but it'd certainly be more optimal to use a map representation instead for reduced runtime complexity.

Reply:
Yep, i was gonna suggest that, but didn't think there were that many logs and they're also filtered. But yea, if its gonna be big, map check would be better.